src/category_theory/limits/shapes/hom_equiv/binary_products.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.limits.shapes.binary_products
+
+/-!
+Constructing binary products from specified objects, and characterisations of the morphisms
```

Comment:
Could you expand the module doc to comply with https://github.com/leanprover-community/mathlib/blob/master/docs/contribute/doc.md?

Reply:
Thanks for the reminders about documentation. I want to wait until @rwbarton has a chance to 👍 or 👎 on this one before finishing everything up --- while this approach seems to work reasonably well, I'm not absolutely convinced it's the right way to go yet.